spec/index.html

```diff
@@ -617,6 +633,132 @@ <h2>Streams</h2>
+
+      <p class="issue" data-number="63">
+        <a>YAML streams</a> may correspond more directly to
+        [[[RFC7464]]], which are not presently part of the
+        JSON-LD representation model.
+        The description here more closely aligns with how JSON-LD
+        interprets <a data-cite="JSON-LD11-API#html-content-algorithms">HTML Scripts</a>.
+      </p>
+      <ol>
+        <li>Set <var>stream content</var> to an empty <a>array</a>.</li>
+        <li>If the <a>stream</a> is empty,
+          set <var>stream content</var> to an empty <a>array</a>.</li>
+        <li>Otherwise, if the <a>stream</a> contains a single <a>YAML document</a>,
+          set <var>stream content</var> the result of
+          <a href="#convert-document" class="sectionRef"></a>.</li>
+        <li>Otherwise: for each <var>document</var> in the <a>stream</a>:
+          <ol>
+            <li>Set <var>doc</var> to the result of <a href="#convert-document" class="sectionRef"></a> for <var>document</var>.</li>
+            <li>If <var>doc</var> is an <a>array</a>,
+              merge it to the end of <var>stream content</var>.</li>
+            <li>Otherwise, append <var>doc</var> to <var>stream content</var></li>
+          </ol>
+        </li>
+        <li>The conversion result is <var>stream content</var>.</li>
+      </ol>
+      <p>Any error reported in a recursive processing step results
+        in the failure of this processing step.</p>
+    </section>
+
+    <section id="convert-document">
+      <h3>Converting a <a>YAML document</a></h3>
+
+      <p>From the <a data-cite="YAML#">YAML grammar</a>,
+        a YAML document MAY be proceeded by a
+        <a data-cite="YAML#rule-l-document-prefix">Document Prefix</a>
+        If a document contains a <a data-cite="YAML#rule-c-byte-order-mark">byte order mark</a>, it MUST
+        indicate the UTF-8 character encoding.
+        Otherwise a processing error has been found, and processing is aborted.</p>
+
+      <p>If a document includes one or more <a data-cite="YAML#rule-l-directive">directives</a>, these MUST be processed
+        as per [[YAML]].</p>
+
+      <p>A <a data-cite="YAML#rule-l-bare-document">YAML bare document</a>
+        is composed of a single <a>node</a>.
+        The conversion result is the result of processing that <a>node</a>
+        using the appropriate conversion step defined in this section.
+        If that <a>node</a> is not one of the following,
+        a processing error has been found, and processing is aborted.</p>
+
+      <ul>
+        <li><a>YAML block sequence</a>,</li>
+        <li><a>YAML flow sequence</a>,</li>
+        <li><a>YAML block mapping</a>, or</li>
+        <li><a>YAML flow mapping</a>.</li>
+      </ul>
+      <p>Any error reported in a recursive processing step results
+        in the failure of this processing step.</p>
+    </section>
+
+    <section id="convert-block-seq">
+      <h3>Converting a <a>YAML block sequence</a></h3>
+    </section>
```

Comment:
Yes, it is tedious. It would be better to rely on an external definition of such a transformation, since other than the use of the internal representation, this is fairly generic YAML processing.

@ioggstream has suggested raising an issue on the YAML media-type repo, although it's not clear that that is a proper place for it either.

The line I'm going on here is still relatively simple, but making sure that anchors, references, and tags are properly treated may push it too far.

Reply:
We will/should of course have comprehensive tests for these things, but we can't rely on tests to specify normative behavior. We need unambiguous description of the transformation process, at least into our _internal representation_.

That said, I'd much rather refer to a process than specify it (see https://github.com/ietf-wg-httpapi/mediatypes/issues/59).